apps/src/dance/ai/DanceAiClient.ts

```diff
@@ -112,23 +110,24 @@ export function getGeneratedEffectScores(
 /**
  * Looks up and sums element-wise precalculated vector weights for each input as a condensed final output
  * Emulates summation process for final classifier output layer of a machine learning model
- * @param {*} emojis, associated input keywords that the model has precalculated
- * @param {*} associatedOutputJson, precalculated vector weights for each possible type of output (e.g. BackgroundsEffects, ForegroundEffects, etc.)
- * @returns 1d array of values between [-3, 3] with indexes that map to output "classes" (e.g. "circles" for BackgroundEffects) Higher values = stronger correlation
+ * @param {*} emojis, set of three emojis selected by user
+ * @param {*} outputWeightsMapping, precalculated association weights for a possible type of output (e.g. background effect, foreground effect or palette)
+ * @returns 1d array of values between [0, 3] with indexes that map to output "option" (e.g. "circles" for BackgroundEffects) Higher values = stronger correlation
  */
-function calculateOutputWeightsVector(
+function calculateOutputSummedWeights(
   emojis: string[],
-  associatedOutputJson: CachedWeightsMapping
+  outputWeightsMapping: CachedWeightsMapping
 ) {
-  const individualInputVectors: number[][] = emojis.map(emojiName => {
-    return associatedOutputJson['emojiAssociations'][emojiName];
+  // selectedEmojiAssociations is an array of 3 subarrays. Each subarray contains
+  // the weights ranging from [0, 1] for the selected emojis for a particular output type)
```

Comment:
Don't have to, but if we want to get super strict about this, we can enforce these length requirements with TS directly - something like `type EmojiAssociations = [number[], number[], number[]]`

Reply:
I tried `[number, number, number]` but got back the error:
'Type 'number[]' is not assignable to type '[number, number, number]'.
  Target requires 3 element(s) but source may have fewer.'
  I tried changing type of param `emojis` to `[string, string, string]` but error still persisted for some reason. So changed back return type back to `number[]`.